packages/components/src/viewport/aware/index.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+import React from 'react';
+import { View } from 'react-native';
+import * as Utils from '../utils'
+
+export default WrappedComponent => {
+  return class extends React.Component {
+    constructor(props, context) {
+      super(props, context);
+      this.state = {
+        mounted: false,
+        inViewport: false,
+      };
+    }
+
+    componentDidMount() {
+      this.context.addViewportListener
+        && this.context.addViewportListener(this._onViewportChange);
+      this.setState({
+        mounted: true,
+      });
+    }
+
+    componentWillUnmount() {
+      this.context.removeViewportListener
+        && this.context.removeViewportListener(this._onViewportChange);
+    }
+
+    _onViewportChange = (info) => {
+      if (this.state.mounted) {
+        this.wrapperRef.measureLayout(info.parentHandle, (wrapperX, wrapperY, wrapperWidth, wrapperHeight) => {
+          const inViewport = Utils.isInViewport(
+            info.viewportOffset, info.viewportHeight, wrapperY, wrapperHeight, this.props.preTriggerRatio);
+          this.setState({
+            inViewport,
+          });
+        });
+      }
+    };
+
+    render() {
+      return (
+        <View ref={ref => this.wrapperRef = ref}>
```

Comment:
What about style attributes of the wrapped componebnt? do you really need this View?

Reply:
It would have been perfect if we could remove it, but yes, I think we really need this `View`.

If we put the `ref` on the wrapped component we get an error: `_this.wrapperRef.measureLayout is not a function`. This happens when `_onViewportChange` is initially called.

I've also had some other problems with this approach in the past: https://github.com/nzzdev/nzz_vamp_engine/pull/27#discussion_r122949800.

I suggest that we keep the `View` after all.